### ISSUE TITLE
main: detached scanBg thread outlives allocator lifetime

### ISSUE THREAD

**Issue by justrach** [justrach/codedb#42]
[Problem]
The scanBg function is spawned with .detach() and receives allocator, data_dir, abs_root from main's GPA. When main returns, deferred cleanup (gpa.deinit(), allocator.free(data_dir)) runs while detached scanBg may still access them—use-after-free crash. MCP startup joins watch_thread, isr_thread, idle_thread but NOT scan_thread.

Relevant lines:
- `src/main.zig:437-438` — spawn + detach of scan_thread
- `src/main.zig:541` — scanBg signature (receives allocator, data_dir, abs_root)
- `src/main.zig:449-451` — watch/isr/idle joined, scan_thread absent

[Failing Test]
```zig
test "issue-42: detached scanBg thread outlives allocator lifetime" {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const allocator = gpa.allocator();
    const data_dir = try allocator.dupe(u8, "/tmp/codedb_test_issue42");

    const SharedCtx = struct {
        data_dir: []u8,
        allocator: std.mem.Allocator,
        done: std.atomic.Value(bool),

        fn run(ctx: *@This()) void {
            std.Thread.sleep(10 * std.time.ns_per_ms);
            _ = ctx.data_dir[0]; // UAF if allocator freed by main
            ctx.done.store(true, .release);
        }
    };

    var ctx = SharedCtx{
        .data_dir = data_dir,
        .allocator = allocator,
        .done = std.atomic.Value(bool).init(false),
    };

    const scan_thread = try std.Thread.spawn(.{}, SharedCtx.run, .{&ctx});
    scan_thread.detach(); // mirrors main.zig:438

    allocator.free(data_dir); // UAF: thread still holds pointer
    _ = gpa.deinit();         // frees GPA while thread may be using it

    try testing.expect(ctx.done.load(.acquire)); // FAILS: thread not done yet
}
```

[Expected]
Main should not deinit memory while scanBg is running.

[Fix]
Don't detach scanBg—store thread handle and join before main exits, or use atomic shutdown flag.
